src/js/tools/type.js

```diff
@@ -124,7 +126,15 @@ define(function (require, exports, module) {
                 var documentStore = this.flux.store("application"),
                     currentDocument = documentStore.getCurrentDocument();
 
-                this.flux.actions.layers.resetBounds(currentDocument, currentDocument.layers.allSelected);
+                // Debounce to allow history state event to precede the call to resetBounds
+                var debouncedMoveHandler = synchronization.debounce(function () {
```

Comment:
This is a little scary, but... OK? I guess it's just swapping one frightening thing for a different one.